### ISSUE TITLE
training G funstion on imagenet

### ISSUE THREAD

**Issue by sesmae** [GT-RIPL/L2C#7]
Hi, I have question regarding training function G. by using pair-enumeration layer there are always many more dis-similar pair per batch than similar pairs, I can see that it works on Omniglot but on more complicated datasets like imagenet, would not this imbalance be a problem? is different hyperparameters used to train G for imagenet?

**Comment by yenchanghsu** [comment]
Hi, the Omniglot has more than nine hundred classes, so the situation is the same. The imbalance is addressed by [sampling in the Omniglot dataloader](https://github.com/GT-RIPL/L2C/blob/master/dataloaders/default.py#L107).

Training the G from scratch on ImageNet can be time-consuming. There is a short cut for building the G without training. Here are the steps: 
(1) Pick an ImageNet-pretrained model. 
(2) Feed two images separately to get two predicted probability vectors (softmax outputs). 
(3) The inner product between the two vectors is the probability of being the same class (similar pair). 
(4) You can thresholding the final probability to get the binarized prediction.
You can find more details about this method in [Reference](https://smartech.gatech.edu/bitstream/handle/1853/62814/HSU-DISSERTATION-2020.pdf) Section 6.3.2.

I hope this helps.

**Comment by sesmae** [comment]
The shortcut method seems interesting , I will give it a try, thanks for the hint and for the reference.

**Comment by liujianzhao6328057** [comment]
Hi, I cannot find the code of training function G, could you please upload a version of source code for reference?

**Comment by yenchanghsu** [comment]
This [line](https://github.com/GT-RIPL/L2C/blob/master/scripts/exp_unsupervised_transfer_Omniglot.sh#L2) is the command to train function G.

**Comment by liujianzhao6328057** [comment]
Thanks! I didn't notice it
